Helper/ConfigHelper.php

```diff
@@ -233,32 +168,21 @@ class ConfigHelper
 
      */
     public function __construct(
-        Magento\Framework\App\Config\ScopeConfigInterface $configInterface,
-        StoreManagerInterface                             $storeManager,
-        Currency                                          $currency,
-        DirCurrency                                       $dirCurrency,
-        DirectoryList                                     $directoryList,
-        Magento\Framework\Module\ResourceInterface        $moduleResource,
-        Magento\Framework\App\ProductMetadataInterface    $productMetadata,
-        Magento\Framework\Event\ManagerInterface          $eventManager,
-        SerializerInterface                               $serializer,
-        GroupCollection                                   $groupCollection,
-        GroupExcludedWebsiteRepositoryInterface           $groupExcludedWebsiteRepository,
-        CookieHelper                                      $cookieHelper
-    ) {
-        $this->configInterface = $configInterface;
-        $this->currency = $currency;
-        $this->storeManager = $storeManager;
-        $this->dirCurrency = $dirCurrency;
-        $this->directoryList = $directoryList;
-        $this->moduleResource = $moduleResource;
-        $this->productMetadata = $productMetadata;
-        $this->eventManager = $eventManager;
-        $this->serializer = $serializer;
-        $this->groupCollection = $groupCollection;
-        $this->groupExcludedWebsiteRepository = $groupExcludedWebsiteRepository;
-        $this->cookieHelper = $cookieHelper;
-    }
+        protected Magento\Framework\App\Config\ScopeConfigInterface    $configInterface,
+        protected Magento\Framework\App\Config\Storage\WriterInterface $configWriter,
+        protected StoreManagerInterface                                $storeManager,
+        protected Currency                                             $currency,
+        protected DirCurrency                                          $dirCurrency,
+        protected DirectoryList                                        $directoryList,
+        protected Magento\Framework\Module\ResourceInterface           $moduleResource,
+        protected Magento\Framework\App\ProductMetadataInterface       $productMetadata,
+        protected Magento\Framework\Event\ManagerInterface             $eventManager,
+        protected SerializerInterface                                  $serializer,
+        protected GroupCollection                                      $groupCollection,
+        protected GroupExcludedWebsiteRepositoryInterface              $groupExcludedWebsiteRepository,
+        protected CookieHelper                                         $cookieHelper
+    )
+    {}
```

Comment:
I'm just wondering, is it a whole new way to create constructors in Magento introduced in the last past 3 years ? It's so cool not to have to do it the "old way" 😃

Reply:
@damcou Agree! [Constructor property promotion](https://php.watch/versions/8.0/constructor-property-promotion) is one of my favorite new features in PHP 8 and cleans up this code a lot.